### PR TITLE
HTTP - server metatype

### PR DESCRIPTION
**Pull request by ceolin**
Add support to HTTP server metatype. **It still missing support to code generation**.

### CODE REVIEW

File: src/modules/flow/http-server/http-server.c [hunk 1]
```diff
@@ -296,10 +296,10 @@ common_response_cb(void *data, struct sol_http_request *request)
             "Could not serve request: %s", sol_util_strerrora(-r));
 
         sol_http_params_clear(&response.param);
-        r = sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
-            HTTP_HEADER_CONTENT_TYPE, HTTP_HEADER_CONTENT_TYPE_TEXT));
-        if (r < 0)
+        if (!sol_http_param_add(&response.param, SOL_HTTP_REQUEST_PARAM_HEADER(
+            HTTP_HEADER_CONTENT_TYPE, HTTP_HEADER_CONTENT_TYPE_TEXT))) {
             SOL_WRN("could not set response content-type: text/plain: %s", sol_util_strerrora(-r));
+        }
```
Comment (glima) [inline src/modules/flow/http-server/http-server.c]: I guess this one is the list to be changed to return an int, no?

File: src/modules/flow-metatype/http-composed-server/http-composed-server.c [hunk 2]
```diff
@@ -0,0 +1,1126 @@
...
+        p[j]->data_type = port_type_out->base.packet_type->name;
+        p[j]->array_size = 0;
+        p[j]->base_port_idx = j;
+        p[j]->required = false;
+    }
+
+    return 0;
+
+fail_ports_out_desc:
+    if (j > 0) {
```
Comment (glima) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: This looks strange. Say you failed on the second iteration above, you start freeing index one here.

File: src/modules/flow-metatype/http-composed-server/http-composed-server.c [hunk 3]
```diff
@@ -0,0 +1,1126 @@
...
+        sizeof(struct sol_flow_metatype_port_description));
+    sol_vector_init(in,
+        sizeof(struct sol_flow_metatype_port_description));
+
+    r = get_context_tokens(contents, &buffer, &tokens);
+    SOL_INT_CHECK_GOTO(r, < 0, err);
+
+    sol_buffer_init(&composed_type);
+    r = sol_buffer_append_slice(&composed_type, sol_str_slice_from_str("composed:"));
+    SOL_INT_CHECK_GOTO(r, < 0, err);
```
Comment (glima) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: You leak tokens

File: src/modules/flow-metatype/http-composed-server/http-composed-server.c [hunk 4]
```diff
@@ -0,0 +1,1126 @@
...
+
+        port = sol_vector_append(in);
+        if (!port) {
+            free(name);
+            goto err_tokens;
+        }
+
+        port->name = name;
+        port->type = strndup(type_slice.data, type_slice.len);
+        SOL_NULL_CHECK_GOTO(port->type, err_tokens);
```
Comment (glima) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: You leak name/type and vector entry

Comment (glima) [inline src/modules/flow-metatype/http-composed-server/http-composed-server.c]: Oh, this one is right, disregard.

**Comment by ceolin** [comment]
merged with the fixes

**Pull request closed without merging**